pkg/bpf/exporter.go

```diff
@@ -85,6 +88,11 @@ func (e *exporter) SupportedMetrics() SupportedMetrics {
 	}
 }
 
+func (e *exporter) registerBPFStats() {
```

Comment:
Not sure if reversing the dependency is Ok - Can remove this altogether if preferred...

Reply:
> I just thought it might be useful to get the exporter to register what it supports with the stats when it's created to overwrite any assumed default value with the right stats

I'd rather that we remove the dependency between the 2 packages entirely.

The assumptions around defaults aren't awesome either, but they are the lesser of two evils IMO.
It's worse to be mutating global variables at runtime - data races aren't fun, some readers might cache/copy values so never see the "updated" values etc...

If stats has a hard dependency on the bpf metric detection, then the API of the stats package should be modelled that way - it is with the current `bpfSupportedMetrics` but it's not exactly ergonomic

Reply:
ack - I will just get it to stick to the static set of default metrics....